include/avif/internal.h

```diff
@@ -301,7 +301,24 @@ void avifCodecSpecificOptionsDestroy(avifCodecSpecificOptions * csOptions);
 avifResult avifCodecSpecificOptionsSet(avifCodecSpecificOptions * csOptions, const char * key, const char * value); // if(value==NULL), key is deleted
 
 // ---------------------------------------------------------------------------
-// avifCodec (abstraction layer to use different AV1 implementations)
+// avifCodecType (underlying video format)
+
+// Alliance for Open Media video formats that can be used in the AVIF image format.
+typedef enum avifCodecType
+{
+    AVIF_CODEC_TYPE_UNKNOWN,
+    AVIF_CODEC_TYPE_AV1,
+#if defined(AVIF_CODEC_AVM)
+    AVIF_CODEC_TYPE_AV2, // Experimental.
+#endif
+    AVIF_CODEC_TYPE_COUNT
```

Comment:
Don't define this `AVIF_CODEC_TYPE_COUNT` constant unless it is needed.

Reply:
Done. It was to avoid the dangling comma but trailing commas seem allowed in C99.

Reply:
I see. An alternative is to define the `AVIF_CODEC_TYPE_AV2` enum constant unconditionally, similar to how we define `AVIF_CODEC_CHOICE_AVM` in avif.h.

I don't know whether trailing commas are allowed in C99. If Clang, GCC, and MSVC all compile the new code successfully, then we can considered them as supported in practice.

Reply:
Trailing commas were an addition to C99.

Reply:
https://en.cppreference.com/w/c/language/enum

*...where enumerator-list is a comma-separated list (with trailing comma permitted) (since C99) of enumerator, each of which has the form:...*

Reply:
Thank you, James.

Perhaps it's the initializer list for a struct or an array that may not have a trailing comma in C99?

Reply:
C90 supported that. C99 added array and struct designators.

https://en.cppreference.com/w/c/language/array_initialization